lib/tapioca/compilers/dsl/active_record_columns.rb

```diff
@@ -347,6 +347,25 @@ def do_not_generate_strong_types?(constant)
           Object.const_defined?(:StrongTypeGeneration) &&
               !(constant.singleton_class < Object.const_get(:StrongTypeGeneration))
         end
+
+        def handle_unknown_type(column_type)
+          return "T.untyped" unless column_type.ancestors.include?(ActiveModel::Type::Value)
+
+          lookup_return_type_of_method(column_type, :deserialize) ||
+            lookup_return_type_of_method(column_type, :cast) ||
+            lookup_arg_type_of_method(column_type, :serialize) ||
+            "T.untyped"
+        end
+
+        def lookup_return_type_of_method(column_type, method)
+          signature = T::Private::Methods.signature_for_method(column_type.instance_method(method))
+          signature.return_type.to_s if signature && !signature.return_type.to_s.include?("T::Private::Types")
```

Comment:
This might not be the best way to test for `Void` and `NotTyped`. Can we test for them explicitly please? If you don't want to test by doing type comparison, you can also test the `.to_s` output being `<VOID>` or `<NOT-TYPED>`.